src/contracts/Repository/Values/Content/URLWildcard/Query/SortClause.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Core\Repository\Values\Content\URLWildcard\Query;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+
+/**
+ * This class is the base for SortClause classes, used to set sorting of URL queries.
+ */
+abstract class SortClause
+{
+    public const SORT_ASC = 'ascending';
+    public const SORT_DESC = 'descending';
+
+    /** @var string */
+    public $direction = self::SORT_ASC;
+
+    /** @var string */
+    public $target;
+
+    /**
+     * @param string $sortDirection one of SortClause::SORT_ASC|SortClause::SORT_DESC
```

Comment:
```suggestion
     * @phpstan-param SortClause::SORT_* $sortDirection
```
the credits go to @Steveb-p for that ;)